### PR TITLE
Iroha custom ISI guide and Cucumber tests.

### PR DESCRIPTION
**Pull request by humb1t**
Signed-off-by: Nikita Puzankov <humb1t@yandex.ru>

### CODE REVIEW

File: docs/source/tutorials/mint-your-first-asset.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+```toml
+iroha = "2.0.0"
+```
+
+### 2. Find a place where you need to mint an asset
+
+Usually any business operation will have a need to mint an asset:
+
+* Uploading of digital documents
+* Withdraw of crypto currencies
```
Comment (eadventurous) [inline docs/source/tutorials/mint-your-first-asset.md:27]: `Withdrawal`

Comment (eadventurous) [reply docs/source/tutorials/mint-your-first-asset.md:27]: And also why withdrawal? I would add generating an initial supply of cryptocurrency as example.

File: docs/source/tutorials/mint-your-first-asset.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+If not - you will need to receive Account's Key Pair with permissions to Register an Account.
+
+TL;DR - after [configuration of Iroha CLI](https://github.com/hyperledger/iroha/blob/iroha2-dev/iroha_client_cli/README.md)
+run this command:
+
+```bash
+./iroha_client_cli account register --domain="my_domain" --name="my_account" --key="{account_public_key}"
+```
+
+### 4. Construct Iroha Special Instruction
```
Comment (eadventurous) [inline docs/source/tutorials/mint-your-first-asset.md:47]: I think it should be mentioned that we also can submit mint ISI via CLI Client, but the purpose here is to show how to do it with code

File: iroha/src/asset.rs [hunk 3]
```diff
@@ -525,6 +525,63 @@ pub mod query {
             }))
         }
     }
+
+    /// To get the state of all assets in an account filtered by assets definition,
+    /// GetAccountAssetsWithDefinition query can be used.
+    #[derive(Clone, Debug, Io, IntoQuery, Encode, Decode)]
+    pub struct GetAccountAssetsWithDefinition {
```
Comment (eadventurous) [inline iroha/src/asset.rs:532]: I think it is better to add new queries in a separate PR.

**Comment by eadventurous** [comment]
It would be great to have it in 3 PRs:
- cucumber iroha client scenario
- iroha ISI tutorial
- new queries

File: iroha/src/asset.rs (hunk 3 above)
Comment (humb1t) [reply iroha/src/asset.rs:532]: But tests will not work without them

Comment (eadventurous) [reply iroha/src/asset.rs:532]: Ah, I see.

**Pull request merged**